Code/src/main/java/nl/utwente/viskell/ui/components/InputAnchor.java

```diff
@@ -103,6 +103,10 @@ public boolean hasConnection() {
     public Type getType() {
         return this.type;
     }
+    
+    public void setType(Type t) {
```

Comment:
This is not nice indeed, but required for now. We should consider adding a 'inverse binder' to each inputAnchor as a cleaner solution for input type restrictions (however that can wait till after the model split).